### PR TITLE
fix APIs, update conv, fc, pool op for dygraph, test=develop

### PR DESCRIPTION
**Pull request by JepsonWong**
update conv, fc, pool op english APIs for dygraph.

### CODE REVIEW

**Comment by JepsonWong** [comment]
TreeConv
Pool2D
FC
Conv2DTranspose
Conv2D

File: python/paddle/fluid/dygraph/nn.py [hunk 1]
```diff
@@ -2761,47 +2818,48 @@ def forward(self, weight):
...
-        ***Tree-Based Convolution Operator***
-
-        Tree-Based Convolution is a kind of convolution based on tree structure.
-        Tree-Based Convolution is a part of Tree-Based Convolution Neural Network(TBCNN),
-        which is used to classify tree structures, such as Abstract Syntax Tree.
-        Tree-Based Convolution proposed a kind of data structure called continuous binary tree,
-        which regards multiway tree as binary tree.
-        The paper of Tree-Based Convolution Operator is here: https://arxiv.org/abs/1409.5718v1
-
+    ***Tree-Based Convolution Operator***
```
Comment (JiabinYang) [inline python/paddle/fluid/dygraph/nn.py]: seems it cause preview format error, double check it

File: python/paddle/fluid/dygraph/nn.py [hunk 2]
```diff
@@ -84,48 +86,52 @@ class Conv2D(layers.Layer):
...
-        groups (int): The groups number of the Conv2d Layer. According to grouped
+            dilation_H = dilation_W = dilation. Default: 1.
+        groups (int, optional): The groups number of the Conv2d Layer. According to grouped
             convolution in Alex Krizhevsky's Deep CNN paper: when group=2,
             the first half of the filters is only connected to the first half
             of the input channels, while the second half of the filters is only
-            connected to the second half of the input channels. Default: groups=1.
-        param_attr (ParamAttr|None): The parameter attribute for learnable parameters/weights
+            connected to the second half of the input channels. Default: 1.
+        param_attr (ParamAttr, optional): The parameter attribute for learnable parameters/weights
```
Comment (JiabinYang) [inline python/paddle/fluid/dygraph/nn.py]: what do u mean by parameters/weights? Maybe change to "weights(Parameter)"

File: python/paddle/fluid/dygraph/nn.py [hunk 3]
```diff
@@ -84,48 +86,52 @@ class Conv2D(layers.Layer):
...
             of conv2d. If it is set to None or one attribute of ParamAttr, conv2d
             will create ParamAttr as param_attr. If the Initializer of the param_attr
             is not set, the parameter is initialized with :math:`Normal(0.0, std)`,
             and the :math:`std` is :math:`(\\frac{2.0 }{filter\_elem\_num})^{0.5}`. Default: None.
-        bias_attr (ParamAttr|bool|None): The parameter attribute for the bias of conv2d.
+        bias_attr (ParamAttr or bool, optional): The parameter attribute for the bias of conv2d.
```
Comment (JiabinYang) [inline python/paddle/fluid/dygraph/nn.py]: "The parameter attribute" -> "The attribute"

File: python/paddle/fluid/dygraph/nn.py [hunk 4]
```diff
@@ -723,33 +729,81 @@ def forward(self, input):
...
-        pool_stride (int|list|tuple): The pool stride size. If pool stride size is a tuple or list,
+            Otherwise, the pool kernel size will be a square of an int. Default: -1.
+        pool_type(str, optional) : The pooling type, can be "max" for max-pooling and "avg" for average-pooling. 
+            Default: max.
+        pool_stride (int or list or tuple, optional): The pool stride size. If pool stride size is a tuple or list,
             it must contain two integers, (pool_stride_Height, pool_stride_Width). Otherwise,
-            the pool stride size will be a square of an int. Default: 1
-        pool_padding (int|list|tuple): The pool padding size. If pool padding size is a tuple,
+            the pool stride size will be a square of an int. Default: 1.
+        pool_padding (int or list or tuple, optional): The pool padding size. If pool padding size is a tuple,
```
Comment (JiabinYang) [inline python/paddle/fluid/dygraph/nn.py]: "The pool padding size" -> "The padding size for pooling operation"

File: python/paddle/fluid/dygraph/nn.py [hunk 5]
```diff
@@ -723,33 +729,81 @@ def forward(self, input):
...
             it must contain two integers, (pool_padding_on_Height, pool_padding_on_Width).
-            Otherwise, the pool padding size will be a square of an int. Default: 0
-        global_pooling (bool): Whether to use the global pooling. If global_pooling = true,
-            kernel size and paddings will be ignored. Default: False
-        use_cudnn (bool): Only used in cudnn kernel, need install cudnn. Default: True
-        ceil_mode (bool): Whether to use the ceil function to calculate output height and width.
-            False is the default. If it is set to False, the floor function will be used. Default: False
-        exclusive (bool): Whether to exclude padding points in average pooling mode. Default: True
+            Otherwise, the pool padding size will be a square of an int. Default: 0.
```
Comment (JiabinYang) [inline python/paddle/fluid/dygraph/nn.py]: same as above, weird syntax

File: python/paddle/fluid/dygraph/nn.py [hunk 6]
```diff
@@ -836,84 +890,86 @@ class FC(layers.Layer):
...
             tensor is flattened: the first `num_flatten_dims` (inclusive, index starts from 1)
             dimensions will be flatten to form the first dimension of the final matrix (height of
             the matrix), and the rest `rank(X) - num_flatten_dims` dimensions are flattened to
             form the second dimension of the final matrix (width of the matrix). For example, suppose
             `X` is a 5-dimensional tensor with a shape [2, 3, 4, 5, 6], and `num_flatten_dims` = 3.
             Then, the flattened matrix will have a shape [2 x 3 x 4, 5 x 6] = [24, 30]. Default: 1
-        param_attr (ParamAttr|list of ParamAttr|None): The parameter attribute for learnable
-            parameters/weights of this layer.
-        bias_attr (ParamAttr|list of ParamAttr, default None): The parameter attribute for the bias
+        param_attr (ParamAttr or list of ParamAttr, optional): The parameter attribute for learnable
```
Comment (JiabinYang) [inline python/paddle/fluid/dygraph/nn.py:955]: same as above

File: python/paddle/fluid/dygraph/nn.py [hunk 7]
```diff
@@ -836,84 +890,86 @@ class FC(layers.Layer):
...
+            parameters/weights of this layer. Default: None.
+        bias_attr (ParamAttr or list of ParamAttr, optional): The parameter attribute for the bias
```
Comment (JiabinYang) [inline python/paddle/fluid/dygraph/nn.py]: same

**Review by JiabinYang — COMMENTED** [review]
some comments

File: python/paddle/fluid/dygraph/nn.py (hunk 2 above)
Comment (JepsonWong) [reply python/paddle/fluid/dygraph/nn.py]: done

File: python/paddle/fluid/dygraph/nn.py (hunk 3 above)
Comment (JepsonWong) [reply python/paddle/fluid/dygraph/nn.py]: done

File: python/paddle/fluid/dygraph/nn.py (hunk 4 above)
Comment (JepsonWong) [reply python/paddle/fluid/dygraph/nn.py]: done

File: python/paddle/fluid/dygraph/nn.py (hunk 5 above)
Comment (JepsonWong) [reply python/paddle/fluid/dygraph/nn.py]: done

File: python/paddle/fluid/dygraph/nn.py (hunk 6 above)
Comment (JepsonWong) [reply python/paddle/fluid/dygraph/nn.py:955]: done

File: python/paddle/fluid/dygraph/nn.py (hunk 7 above)
Comment (JepsonWong) [reply python/paddle/fluid/dygraph/nn.py]: done

**Review by JiabinYang — DISMISSED** [review]
LGTM

**Review by JiabinYang — APPROVED** [review]
GTM

**Review by hutuxian — APPROVED** [review]

**Review by wanghaoshuang — APPROVED** [review]

**Pull request merged**